pkg/adaptor/proxy/proxy.go

```diff
@@ -105,26 +105,27 @@ func (p *agentProxy) dial(ctx context.Context, address string) (net.Conn, error)
 	return conn, nil
 }
 
-func initCriClient(ctx context.Context, target string) (*criClient, error) {
-	if target != "" {
-		conn, err := grpc.DialContext(ctx, target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(),
+func (p *agentProxy) initCriClient(ctx context.Context) (*criClient, error) {
+	if p.criSocketPath != "" {
+		timeout, cancel := context.WithTimeout(ctx, p.retryInterval)
+		defer cancel()
+		conn, err := grpc.DialContext(timeout, p.criSocketPath, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock(),
 			grpc.WithContextDialer(func(ctx context.Context, target string) (net.Conn, error) {
 				return (&net.Dialer{}).DialContext(ctx, "unix", target)
 			}),
 		)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to established cri uds connection to %s", p.criSocketPath)
```

Comment:
Could you also print `retryInterval`?

Reply:
hmm... ok, it depends on the error type, it might not fail due to the timeout.

Reply:
This doesn't have any retry logic, as the socket should either be there or not at this point. Unlike the agent-protocol-forwarder connection.